conans/client/generators/cmake_find_package.py

```diff
@@ -275,17 +254,21 @@ def _find_for_dep(self, pkg_name, pkg_findname, cpp_info):
         pkg_version = cpp_info.version
         pkg_public_deps = [self._get_name(self.deps_build_info[public_dep]) for public_dep in
                            cpp_info.public_deps]
+        pkg_public_deps_names = ";".join(["{n}::{n}".format(n=n) for n in pkg_public_deps])
         if cpp_info.components:
             components = self._get_components(pkg_name, pkg_findname, cpp_info)
             # Note these are in reversed order, from more dependent to less dependent
             pkg_components = " ".join(["{p}::{c}".format(p=pkg_findname, c=comp_findname) for
                                        comp_findname, _ in reversed(components)])
-            pkg = DepsCppCmake(cpp_info)
+            pkg_info = DepsCppCmake(cpp_info)
+            global_target_variables = target_template.format(name=pkg_findname, deps=pkg_info,
+                                                             build_type_suffix="",
+                                                             deps_names=pkg_public_deps_names)
```

Comment:
If this `deps_name` is just the list of the components contained in this package we get rid of the direct dependency to the global target of the requirements (`pkg_components`?).

Reply:
No, not really needed, this is not populating the target, it is just about the global variables. So it can be exactly like the other branch without components. 👍